build.gradle

```diff
@@ -79,30 +75,37 @@ task wrapper(type: Wrapper) {
     gradleVersion = '4.10.2'
 }
 
-publishing {
-	publications {
-		maven(MavenPublication) {
-			groupId project.group
-			artifactId project.archivesBaseName
-			version project.version
+gradlePlugin {
+    plugins {
+        register("fabricLoom") {
+            id = "fabric-loom"
+            implementationClass = "net.fabricmc.loom.LoomGradlePlugin"
+        }
+    }
+}
 
-			from components.java
+publishing {
+    publications {
+        pluginMaven(MavenPublication) { publication ->
+            groupId project.group
+            artifactId project.archivesBaseName
+            version project.version
 
-			//Removes all of the dependencies from the maven pom, prevents sub projects downloading all the libs, as we use a fat jar
-			pom.withXml {
-				asNode().remove(asNode().get('dependencies'))
-			}
-		}
-	}
-	repositories {
-		maven {
-			url "http://mavenupload.modmuss50.me/"
-			if (project.hasProperty('mavenPass')) {
-				credentials {
-					username 'buildslave'
-					password project.getProperty('mavenPass')
-				}
-			}
-		}
-	}
-}
+            //Removes all of the dependencies from the maven pom, prevents sub projects downloading all the libs, as we use a fat jar
+            pom.withXml {
+                asNode().remove(asNode().get('dependencies'))
+            }
+        }
+    }
+    repositories {
+        maven {
+            url "http://mavenupload.modmuss50.me/"
+            if (project.hasProperty('mavenPass')) {
+                credentials {
+                    username 'buildslave'
+                    password project.getProperty('mavenPass')
+                }
+            }
+        }
+    }
+}
```

Comment:
The newline should be retained, as should the tabs :)

It would seem that the buildscript currently uses a mix of tabs and spaces, so doing a format here could make sense :)

Reply:
i added a .editorconfig and reverted it to tabs only

Reply:
Still lacking newline at the end..?